spec/controllers/photos/index_spec.rb

```diff
@@ -0,0 +1,37 @@
+# coding: utf-8
+require 'rails_helper'
+
+RSpec.describe PhotosController, type: :controller do
+  # pending "add some examples to (or delete) #{__FILE__}"
+
+  before(:each) do
+    @photo_house = create(:photo_house)
+    @photo_mountains = create(:photo_mountains)
+    @photo_waterfall = create(:photo_waterfall)
+    @photo_way = create(:photo_way)
+
+    get :index
+    
+    @photolist = controller.instance_variable_get("@photos")
+  end
+
+  it "200レスポンスが返されている。" do
+    expect(response).to have_http_status "200"
+  end
+
+  it "モデルからデータを全て（本テストでは４つ）取得出来ている。" do
+    expect(@photolist.length).to eq 4
+  end
+
+  it "モデルから取得したデータがデータベース登録降順になっている。" do
+    original_created_at = [@photo_waterfall.created_at, @photo_way.created_at, @photo_house.created_at, @photo_mountains.created_at]
+
+    @data = []
+    
+    @photolist.each do |photo|
+      @data.push(photo.created_at)
+    end
```

Comment:
```suggestion
    @data = @photolist.map(&:created_at)
```
他のテストについても同様

どういうことをしているかは確認しておいて下さい
https://qiita.com/k-penguin-sato/items/420d7487b28b5d58cac4